magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/UserController.php

```diff
@@ -321,29 +323,61 @@ public function meAction(){
+                                        'vsbridge.log'
+                                    );
+                                }
+                            } elseif ($updatedAdress['default_billing']) {
                                 $bAddressId = $customer->getDefaultBilling();
                                 $bAddress = Mage::getModel('customer/address');
 
-                                if($bAddressId)
-                                    $bAddress->load($bAddressId);
+                                if ($bAddressId) $bAddress->load($bAddressId);
 
                                 $updatedAdress['parent_id'] = $customer->getId();
-
+                                $updatedAdress['street'] = $addressHelper->concatStreetData($updatedAdress['street']);
 
                                 $bAddress->setData($updatedAdress)->setIsDefaultBilling(1)->save();
                                 $updatedBillingId = $bAddress->getId();
-                            }
-                            if($updatedAdress['default_shipping']) {
+                            } elseif ($updatedAdress['default_shipping']) {
                                 $sAddressId = $customer->getDefaultShipping();
                                 $sAddress = Mage::getModel('customer/address');
 
-                                if($sAddressId)
-                                    $sAddress->load($sAddressId);
+                                if ($sAddressId) $sAddress->load($sAddressId);
 
                                 $updatedAdress['parent_id'] = $customer->getId();
+                                $updatedAdress['street'] = $addressHelper->concatStreetData($updatedAdress['street']);
+
                                 $sAddress->setData($updatedAdress)->setIsDefaultShipping(1)->save();
                                 $updatedShippingId = $sAddress->getId();
+                            } else {
+                                $sAddress = Mage::getModel('customer/address');
+
+                                try {
+                                    if (isset($updatedAdress['entity_id']) && !empty($updatedAdress['entity_id'])) {
+                                        $sAddress->load($updatedAdress['entity_id']);
+                                    }
+                                } catch (Exception $exception) {
+                                    Mage::log(
+                                        'Cloud not load address:' . $updatedAdress['entity_id'] . " for updating it.",
+                                        Zend_Log::ERR,
+                                        'vsbridge.log'
+                                    );
+                                }
+
+                                $updatedAdress['parent_id'] = $customer->getId();
+                                $updatedAdress['street'] = $addressHelper->concatStreetData($updatedAdress['street']);
+
+                                $sAddress->setData($updatedAdress)->save();
```

Comment:
I'm not sure I understand this save action, isn't the address saved in previous steps in this class?

Reply:
This should now be fixed.